app/src/main/java/org/mozilla/tv/firefox/FirefoxApplication.kt

```diff
@@ -110,15 +110,17 @@ open class FirefoxApplication : LocaleAwareApplication() {
                 // sending startup data, or 2) sending even when the user has toggled
                 // off data collection
                 Glean.setUploadEnabled(collectionEnabled)
+                if (collectionEnabled) {
```

Comment:
Is this code here actually needed? The legacy client id is already set in `initGlean` - why is that not enough?

I don't think this code will actually do change anything. If collectionEnabled is true then the legacy id will be set to the exact same value as in `initGlean` already did? What do I miss?

Reply:
@st3fan so the problem is that when the `collectionEnabled` is set to false on app start, `LegacyId.clientId` is not set (which I think is by design because you don't want to send metric pings when `collectionEnabled` is false). So as of now, if `collectionEnabled` is false on app start, then setting the `collectionEnabled` to true won't guarantee `LegacyId.clientId` to be set until you restart the app.

The code block here is observed stream so whenever collectionEnabled is set to true, we make sure the `LegacyId.clientId` is set. Better way to do this is to have `SharedPreference` hold an one-time flag for `LegacyId.clientId` and have the condition like:
```
if (collectionEnabled && !legacyIdSet) {
```

There is an edge case where the `collectionEnabled` is false from the beginning and never set to true. This would theoretically have missing `LegacyId.clientId` as well.